ui/src/store/SignalHandler.ts

```diff
@@ -17,24 +17,29 @@ export function createSignalHandler(
 
   async function _handleSignalReceived(signal: Signal) {
     if (!(SignalType.App in signal)) return;
-    if ((signal[SignalType.App].payload as RelaySignal).type !== "Message") return;
 
-    // Ignore signals for messages I sent
-    if (isEqual((signal[SignalType.App].payload as MessageSignal).from, client.client.myPubKey))
-      return;
-
-    // Save recieved message
+    const payload = signal[SignalType.App].payload as RelaySignal;
     const cellIdB64 = encodeCellIdToBase64(signal[SignalType.App].cell_id);
-    await conversationMessageStore.handleMessageSignalReceived(
-      cellIdB64,
-      signal[SignalType.App].payload as MessageSignal,
-    );
 
-    // Mark conversation as unread
-    // Unless user is currently viewing the conversation page.
-    const $page = get(page);
-    if ($page.params.id !== cellIdB64 || $page.route.id !== "/conversations/[id]") {
-      await conversationStore.updateUnread(cellIdB64, true);
+    if (payload.type === "Message") {
+      // Ignore signals for messages I sent
+      if (isEqual(payload.from, client.client.myPubKey)) return;
+      await conversationMessageStore.handleMessageSignalReceived(
+        cellIdB64,
+        signal[SignalType.App].payload as MessageSignal,
+      );
+      // Mark conversation as unread
+      // Unless user is currently viewing the conversation page.
+      const $page = get(page);
+      if ($page.params.id !== cellIdB64 || $page.route.id !== "/conversations/[id]") {
+        await conversationStore.updateUnread(cellIdB64, true);
+      }
+    } else if (payload.type === "EntryDeleted" && payload.original_app_entry.type === "Message") {
+      const originalActionHashB64 = payload.action.hashed.hash;
```

Comment:
```suggestion
      const originalActionHash = payload.action.hashed.hash;
```

Is the actually the original action hash? Or is it the latest action hash (i.e. the delete action)? I think this might be your problem.